src/shared-validator.cc

```diff
@@ -282,7 +284,7 @@ Result SharedValidator::OnElemSegmentElemType(const Location& loc,
   if (elem.is_active) {
     // Check that the type of the elem segment matches the table in which
     // it is active.
-    result |= CheckType(loc, elem.table_type, elem_type, "elem segment");
+    result |= CheckType(loc, elem_type, elem.table_type, "elem segment");
```

Comment:
I just found this bug. The actual/expected arguments are swapped. The tests are updated accordingly.

Reply:
Is this is pre-existing bug?  If so maybe it can be split out into a separate PR?